src/ram/transform/IndexedInequality.cpp

```diff
@@ -112,10 +112,17 @@ bool IndexedInequalityTransformer::transformIndexToFilter(RamProgram& program) {
                                 std::make_unique<RamRelationReference>(&ichoice->getRelation()),
                                 ichoice->getTupleId(), souffle::clone(&ichoice->getCondition()),
                                 std::move(updatedPattern), std::move(filter), ichoice->getProfileText());
+                        // in the case of an aggregate we must strengthen the condition of the aggregate
```

Comment:
Put this comment inside the agg section. It looks like it's part of the index choice section as it is.